source/fundamentals/crud/compound-operations.txt

```diff
@@ -0,0 +1,293 @@
+
+Overview
+--------
+
+On this page you will learn how to perform **compound operations** with the
+MongoDB Java driver. 
+
+Compound operations consist of a read and write operation performed as one
+**atomic operation**. An atomic operation is an irreducible series of one or
+more database operations such that all occur or none occur.
+
+Atomic operations can help avoid certain kinds of **race conditions** in your code. A
+race condition occurs when your code's behavior is dependent on the order of
+uncontrollable events.
+
+MongoDB supports the following compound operations:
+
+- Find and update one document
+- Find and replace one document
+- Find and delete one document
+
+If you need to perform more complex tasks atomically, such as reading and
+writing to more than one document, use **transactions**.
+
+Transactions are a feature of MongoDB and other databases that lets you define
+an arbitrary sequence of database commands as an atomic operation.  
+
+For more information on atomic operations and atomicity, see 
+:manual:`the MongoDB manual entry for atomicity and transactions </core/write-operations-atomicity/>`.
+
+For more information on transactions, 
+:manual:`see the MongoDB manual entry for transactions </core/transactions/>`.
+
+How to Use Compound Operations
+------------------------------
+
+This section shows how to use each compound operation with the MongoDB Java Driver.
+
+The following examples use a collection containing these two sample documents.
+
+.. code-block:: json
+
+    {"_id": 1, "food": "donut", "color": "green"}
+    {"_id": 2, "food": "pear", "color": "yellow"}
+
+`The full code for the following examples is available on Github here. <https://github.com/mongodb/docs-java/blob/master/source/includes/fundamentals/code-snippets/CompoundOperatorsIndividualExamples.java>`__
+
+Find and Update One Document
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To find and update a document, use the ``findOneAndUpdate()`` method of the
+``MongoCollection`` class.
+
+The following example uses the ``findOneAndUpdate()`` method to find a
+document with the ``color`` field set to ``"green"`` and  update the
+``food`` field in that document to ``"pizza"``. 
+
+The example also uses a ``FindOneAndUpdateOptions`` instance to specify the
+following options:
+
+- Exclude the ``_id`` field from the found document
+- Specify to insert a document if no documents match our query. MongoDB and
+  other databases refer to this operation as an upsert.
+- Set a maximum execution time of 5 seconds for this operation on the MongoDB
+  instance. If the operation takes longer, the ``findOneAndUpdate()`` method
+  will throw a ``MongoExecutionTimeoutException``.  
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CompoundOperatorsIndividualExamples.java
+   :language: java
+   :dedent:
+   :start-after: start findOneAndUpdate-example
+   :end-before: end findOneAndUpdate-example
+
+The output of the above code should look like this:
```

Comment:
Suggestion: I think it would be beneficial to explain to the reader what your code is printing.

Applies to all instances.

Reply:
Thanks for the note. I think the updates from https://github.com/mongodb/docs-java/pull/80#discussion_r632748711 (adding description of return value) solve this issue.